src/validators/db-lookup.validator.ts

```diff
@@ -19,28 +19,53 @@ export class DbLookupValidator implements ValidatorConstraintInterface {
       findOption,
     }: DbLookupOptions<BaseEntity> = args.constraints[0];
 
+    // Handle empty or null values
     if (value || !ignoreEmpty) {
+      // Additional check for non-numeric input (assuming IDs are numeric)
+      if (isNaN(value)) {
+        throw new BadRequestException(`Invalid input: "${value}" is not a valid numeric identifier.`);
+      }
```

Comment:
This error is incorrectly thrown when a string value is passed in for validation. If this validation is desired, it should be applied only to inputs that are _supposed_ to be a number.

Reply:
``` markdown
Thanks, MaxD! I have made the following changes on < db-lookup.validator.ts> to correct the error you mentioned. 
I have added optional numeric validation with `validateNumeric` flag, improved the error handling and added proper type checking.

Reply:
``` markdown
On the <.idea/ files>, I was an oversight as I rushing to push before our meeting. Thx.